Sources/MapboxNavigation/MapView.swift

```diff
@@ -174,33 +174,94 @@ extension MapView {
             NavigationMapView.LayerIdentifier.arrowLayer,
-            NavigationMapView.LayerIdentifier.arrowSymbolLayer,
             NavigationMapView.LayerIdentifier.arrowSymbolCasingLayer,
-            NavigationMapView.LayerIdentifier.arrowStrokeLayer,
+            NavigationMapView.LayerIdentifier.arrowSymbolLayer
+        ].compactMap{ $0 }
+        let aboveSymbolLayers: [String] = [
+            NavigationMapView.LayerIdentifier.buildingExtrusionLayer,
             NavigationMapView.LayerIdentifier.waypointCircleLayer,
-            NavigationMapView.LayerIdentifier.buildingExtrusionLayer
+            NavigationMapView.LayerIdentifier.waypointSymbolLayer,
+            NavigationMapView.LayerIdentifier.continuousAlternativeRoutesDurationAnnotationsLayer,
+            NavigationMapView.LayerIdentifier.routeDurationAnnotationsLayer,
+            NavigationMapView.LayerIdentifier.voiceInstructionLabelLayer,
+            NavigationMapView.LayerIdentifier.voiceInstructionCircleLayer,
+            NavigationMapView.LayerIdentifier.puck2DLayer,
+            NavigationMapView.LayerIdentifier.puck3DLayer
         ]
+        let belowSymbol = !aboveSymbolLayers.contains(layerIdentifier)
+        let allAddedLayers: [String] = belowSymbolLayers + aboveSymbolLayers
         
-        for layer in mapboxMap.style.allLayerIdentifiers.reversed() {
-            if !(layer.type.rawValue == "symbol") && !identifiers.contains(layer.id) {
-                let sourceLayer = mapboxMap.style.layerProperty(for: layer.id, property: "source-layer").value as? String
-                
-                if let sourceLayer = sourceLayer,
-                   sourceLayer.isEmpty {
-                    continue
-                }
-                
-                parentLayer = layer.id
+        var layerPosition: MapboxMaps.LayerPosition? = nil
+        var lowerLayers = Set<String>()
+        var upperLayers = Set<String>()
+        var foundLayer: Bool = false
+        var targetLayer: String? = nil
+        
+        if let indice = allAddedLayers.firstIndex(of: layerIdentifier) {
+            lowerLayers = Set(allAddedLayers.prefix(upTo: indice))
+            if allAddedLayers.indices.contains(indice + 1) {
+                upperLayers = Set(allAddedLayers.suffix(from: indice + 1))
+            }
+        }
+        
+        for layerInfo in mapboxMap.style.allLayerIdentifiers.reversed() {
+            if lowerLayers.contains(layerInfo.id) {
+                // find the topmost layer that should be below the layerIdentifier.
+                layerPosition = .above(layerInfo.id)
                 break
+            } else if upperLayers.contains(layerInfo.id) {
+                // find the bottommost layer that should be above the layerIdentifier.
+                layerPosition = .below(layerInfo.id)
+            } else if belowSymbol {
+                // find the topmost non symbol layer for layerIdentifier in belowSymbolLayers.
+                if !foundLayer,
```

Comment:
It looks like `foundLayer` is only ever true when `targetLayer` is non-nil, so we can eliminate this variable by checking whether `targetLayer` is nil.